docs/architecture/adr-110-e2e-abci++.md

```diff
@@ -0,0 +1,267 @@
+using the synthax that the library understand. 
+The new grammar is below and can be found inside `test/e2e/pkg/grammar/abci_grammar.md` file.
+
+```abnf
+
+Start : CleanStart | Recovery ;
+
+CleanStart : InitChain StateSync ConsensusExec | InitChain ConsensusExec ;
+StateSync : StateSyncAttempts SuccessSync |  SuccessSync ; 
+StateSyncAttempts : StateSyncAttempt | StateSyncAttempt StateSyncAttempts ;
+StateSyncAttempt : OfferSnapshot ApplyChunks | OfferSnapshot ;
+SuccessSync : OfferSnapshot ApplyChunks ; 
+ApplyChunks : ApplyChunk | ApplyChunk ApplyChunks ;  
+
+Recovery :  ConsensusExec ;
+
+ConsensusExec : ConsensusHeights ;
+ConsensusHeights : ConsensusHeight | ConsensusHeight ConsensusHeights ;
+ConsensusHeight : ConsensusRounds Decide Commit | Decide Commit ;
+ConsensusRounds : ConsensusRound | ConsensusRound ConsensusRounds ;
+ConsensusRound : Proposer | NonProposer ; 
+
+Proposer : PrepareProposal ProcessProposal ; 
+NonProposer: ProcessProposal ;
+
+
+InitChain : "<InitChain>" ;
+Decide : "<FinalizeBlock>" ; 
+Commit : "<Commit>" ;
+OfferSnapshot : "<OfferSnapshot>" ;
+ApplyChunk : "<ApplyChunk>" ; 
+PrepareProposal : "<PrepareProposal>" ; 
+ProcessProposal : "<ProcessProposal>" ;
+ 
+ ```
+
+If you compare this grammar with the original one, you will notice that, in addition to vote extensions,  
+`Info` is removed. The reason is that, as explained in the section [CometBFT's expected behaviour](../../spec/abci/abci%2B%2B_comet_expected_behavior.md#valid-method-call-sequences), one of the 
+purposes of the `Info` method is being part of the RPC handling from an external 
+client. Since this can happen at any time, it complicates the 
+grammar.  
+This is not true in other cases, but since the Application does 
+not know why the `Info` is called, we removed 
+it totally from the new grammar. The Application is still logging the `Info` 
+call, but a specific test would need to be written to check whether it happens
+in the right moment. 
+
+The `gogll` library receives the file with the grammar as input, and it generates the corresponding parser and lexer. Specifically, we need to run 
+`gogll pkg/grammar/abci_grammar.md` from `test/e2e/` directory.
+The resulted code is inside the following directories: 
+- `test/e2e/pkg/grammar/lexer`,
+- `test/e2e/pkg/grammar/parser`,
+- `test/e2e/pkg/grammar/sppf`,
+- `test/e2e/pkg/grammar/token`.
+
+Apart from this auto-generated code, we implemented `GrammarChecker` abstraction
+which knows how to use the generated parser and lexer to verify whether a
+specific execution (list of ABCI++ calls logged by the Application while the
+testnet was running) respects the ABCI++ grammar. The implementation and tests 
+for it are inside `test/e2e/pkg/grammar/checker.go` and 
+`test/e2e/pkg/grammar/checker_test.go`, respectively. 
+
+How the `GrammarChecker` works is demonstrated with the test `TestABCIGrammar`
+implemented in `test/e2e/tests/abci_test.go` file. 
+
+```go
+func TestABCIGrammar(t *testing.T) {
```

Comment:
Please add somewhere in this section a couple of sentences to clarify that the grammar represents **one node**, and so it applied to the logs from each node successively.

This can be gleaned from this code, but it's not at all obvious

Reply:
I will do it!

Reply:
I added sentences about this.